src/main/java/com/example/bookstore/service/CartItemServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package com.example.bookstore.service;
+
+import com.example.bookstore.dto.cartitem.CartItemResponseDto;
+import com.example.bookstore.dto.cartitem.CreateCartItemRequestDto;
+import com.example.bookstore.dto.mapper.CartItemMapper;
+import com.example.bookstore.model.Book;
+import com.example.bookstore.model.CartItem;
+import com.example.bookstore.model.ShoppingCart;
+import com.example.bookstore.model.User;
+import com.example.bookstore.repository.cartitem.CartItemRepository;
+import jakarta.persistence.EntityNotFoundException;
+import java.util.List;
+import java.util.Optional;
+import lombok.AllArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class CartItemServiceImpl implements CartItemService {
+    private CartItemRepository cartItemRepository;
+    private CartItemMapper cartItemMapper;
+
+    @Override
+    public CartItemResponseDto save(CreateCartItemRequestDto requestDto, User user) {
+        CartItem cartItem = cartItemMapper.toCartItem(requestDto, user.getId());
+        CartItem savedCartItem = cartItemRepository.save(cartItem);
+        return cartItemMapper.toResponseDto(savedCartItem);
+    }
+
+    @Override
+    public CartItemResponseDto update(Long id, CreateCartItemRequestDto requestDto) {
+        if (!cartItemRepository.existsById(id)) {

```

Comment:
You anyway need oldItem in case it's present, so why make 2 requests to DB instead of 1?
Use Optional and check with isEmpty or isPresent method